### PR TITLE
MOAR COLORZ, FOLKZ!

### PR DESCRIPTION
**Pull request by tiwiz**
[IMAGE ALT TEXT HERE](http://goo.gl/xnMuKO)

> Dove sono i colori?

_Alessandra Amoroso_

### CODE REVIEW

**Comment by rock3r** [comment]
:x: perche' c'e' Alessandra Amoroso -- WTF dude

**Comment by Takhion** [comment]
@rock3r LOL my exact reaction :x:

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java [hunk 1]
```diff
@@ -51,6 +53,17 @@
         EXCLUDE_SESSIONS.add("http://it.droidcon.com/2015/sessions/barcamp-organization-2/");
         EXCLUDE_SESSIONS.add("http://it.droidcon.com/2015/sessions/droidcon-party/");
         EXCLUDE_SESSIONS.add("http://it.droidcon.com/2015/sessions/chiusura/");
+
+        ROOMS_TO_COLORS = new HashMap<>();
```
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: Meh for the diamond operator (will break on JDK 6).

**Comment by tiwiz** [comment]
Because she asks where are the colours, no? :-D

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 1 above)
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: Also you could totally specify initial capacity here.

Comment (tiwiz) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: I could, but I didn't know earlier how many rooms we have... fixing it but... aren't we like compiling against Java 7? o.o

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java [hunk 2]
```diff
@@ -248,6 +262,19 @@ private static void transformJson(
...
     }
 
+    /**
+     * MOAR COLORZ PLS!
+     */
+    private static String getColorFrom(String roomName) {
+        String roomColor = DEFAULT_ROOM_COLOR;
+
+        if (ROOMS_TO_COLORS.containsKey(roomName)) {
+            roomColor = ROOMS_TO_COLORS.get(roomName);
```
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:272]: If you want to be picky, this is more efficient:

``` java
String roomColor = ROOMS_TO_COLORS.get(roomName);
return roomColor != null ? roomColor : DEFAULT_ROOM_COLOR;
```

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java [hunk 3]
```diff
@@ -248,6 +262,19 @@ private static void transformJson(
         sessions.add(session);
     }
 
+    /**
+     * MOAR COLORZ PLS!
+     */
+    private static String getColorFrom(String roomName) {
+        String roomColor = DEFAULT_ROOM_COLOR;
```
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:269]: Nitpick:

``` java
if (roomName == null) {
    return DEFAULT_ROOM_COLOR;
}
String normalizedName = roomName.toLowerCase();
if (ROOMS_TO_COLORS.containsKey(normalizedName) {
    return ROOMS_TO_COLORS.get(normalizedName);
}
return DEFAULT_ROOM_COLOR;
```

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 2 above)
Comment (tiwiz) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:272]: OK, fixing it :-)

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 3 above)
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:269]: LOL @rock3r ~jinx~!

Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:269]: That means that you also store the room names in lowercase in the `HashMap`, of course.

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 1 above)
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: You assume so. But Gradle files aren't mandating it IIRC.

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 3 above)
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:269]: Nah no point in lowercasing the room names, they always come through the same, and we're hacking stuff here :P

**Comment by rock3r** [comment]
@Takhion WHY DID YOU MERGE THIS????? I :x:'d IT

archer-computer-beat-up

Seriously though. This was to be fixed first.

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 1 above)
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: Yes we are

**Comment by Takhion** [comment]
**IMPULSE MERGE ALL THE THINGS!**

**Comment by Takhion** [comment]
To be fair this does what it's supposed to, and we're waaay past doing things TheRightWay™...

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 1 above)
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: I remember Google using Java 7 stuff, like collapsed exceptions and such, somewhere.

**Comment by rock3r** [comment]
When things blow up I will be there looking at you, leaning on a wall, shaking my head with a sad grin and muttering "for fuck sake, I told you...".

Or not. Also yeah this app is becoming more and more the manifestation of The Ferretti Way™ anyway. It's not our fault, I know, but still.

**Comment by Takhion** [comment]
Yeah I deeply hope nobody will ever EVER touch this code again when the event is over.
Seriously.

**Comment by rock3r** [comment]
We might want to delete the branch and pretend this never happened.

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 1 above)
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java:57]: Ok then. Something something JDK rant.

**Comment by Takhion** [comment]
@rock3r :+1: on the coverup

**Pull request merged**